catalog/validator/catalog-validator-metacardduplication/src/main/java/org/codice/ddf/validator/metacard/duplication/DuplicationValidator.java

```diff
@@ -113,119 +124,185 @@ public void setWarnOnDuplicateAttributes(String[] attributeStrings) {
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
 
-    return getReport(reportDuplicates(metacard));
+    return acceptedMetacards;
   }
 
   @Override
-  public void validate(Metacard metacard) throws ValidationException {
-
-    final Optional<MetacardValidationReport> report = validateMetacard(metacard);
-
-    if (report.isPresent()) {
-      final List<String> errors =
-          report
-              .get()
-              .getMetacardValidationViolations()
-              .stream()
-              .filter(
-                  validationViolation ->
-                      validationViolation.getSeverity().equals(ValidationViolation.Severity.ERROR))
-              .map(ValidationViolation::getMessage)
-              .collect(Collectors.toList());
-      final List<String> warnings =
-          report
-              .get()
-              .getMetacardValidationViolations()
-              .stream()
-              .filter(
-                  validationViolation ->
-                      validationViolation
-                          .getSeverity()
-                          .equals(ValidationViolation.Severity.WARNING))
-              .map(ValidationViolation::getMessage)
-              .collect(Collectors.toList());
-
-      String message =
-          String.format("Duplicate data found in catalog for ID {%s}.", metacard.getId());
-      final ValidationExceptionImpl exception = new ValidationExceptionImpl(message);
-      exception.setErrors(errors);
-      exception.setWarnings(warnings);
-      throw exception;
-    }
+  public UpdateRequest process(UpdateRequest input) {
+    List<Entry<Serializable, Metacard>> result = handleDuplicatesEntries(input.getUpdates());
+    return new UpdateRequestImpl(result, input.getAttributeName(), input.getProperties());
   }
 
-  private Set<ValidationViolation> reportDuplicates(final Metacard metacard) {
+  private List<Entry<Serializable, Metacard>> handleDuplicatesEntries(
+      List<Entry<Serializable, Metacard>> updateList) {
+    List acceptedMetacards =
+        updateList
+            .stream()
+            .map(entry -> new SimpleEntry<>(entry.getKey(), validateDuplicates(entry.getValue())))
+            .filter(entry -> entry.getValue() != null)
+            .collect(Collectors.toList());
 
-    Set<ValidationViolation> violations = new HashSet<>();
+    return acceptedMetacards;
+  }
 
-    if (ArrayUtils.isNotEmpty(warnOnDuplicateAttributes)) {
-      ValidationViolation warnValidation =
-          reportDuplicates(
-              metacard, warnOnDuplicateAttributes, ValidationViolation.Severity.WARNING);
-      if (warnValidation != null) {
-        violations.add(warnValidation);
+  private Metacard validateDuplicates(Metacard metacard) {
+    Set<String> rejectedAttributes;
+    Set<String> errorAttributes;
+    Set<String> warningAttributes;
+    String validationMessage = "Duplicates values of existing metacard for attribute(s): %s";
```

Comment:
✏️ I think the wording here is a little awkward

Reply:
I can agree with that. Any suggestions?

Reply:
Maybe removing the `s` in `Duplicates` is enough 😂